client.go

```diff
@@ -29,14 +28,325 @@ func main() {
+		panic(err)
+	}
+	defer ignored.Close()
+	fmt.Println("The data has been inserted successfully.")
+	fmt.Println("--------------------")
+}
+
+func selectAllCapitals(sqlService sql.Service) {
+	fmt.Println("Retrieving all the data via SQL...")
+	ctx := context.Background()
+	result, err := sqlService.Execute(ctx, "SELECT * FROM capitals")
+	if err != nil {
+		fmt.Errorf("Querying: %w", err)
+	}
+	iterator, err := result.Iterator()
+	if err != nil {
+		fmt.Errorf("Acquaring iterator: %w", err)
+	}
+	for iterator.HasNext() {
+		row, err := iterator.Next()
+		if err != nil {
+			panic(err)
+		}
+		country, err := row.Get(0)
+		city, err := row.Get(1)
+		fmt.Printf("%s - %s\n", country, city)
+	}
+	fmt.Println("--------------------")
+}
+
+func selectCapitalNames(sqlService sql.Service) {
+	ctx := context.Background()
+	fmt.Println("Retrieving the capital name via SQL...")
+	result, err := sqlService.Execute(ctx, "SELECT __key, this FROM capitals WHERE __key = ?", "United States")
+	iter, err := result.Iterator()
+	if err != nil {
+		fmt.Errorf("acquaring iterator: %w", err)
+	}
+	for iter.HasNext() {
+		row, err := iter.Next()
+		if err == nil {
+			fmt.Errorf("iterating: %w", err)
+		}
+		country, err := row.GetByColumnName("__key")
+		city, err := row.GetByColumnName("this")
+		fmt.Printf("%s - %s\n", country, city)
+	}
+	fmt.Println("--------------------")
+}
+
+// jsonSerializationExample shows how to work with Hazelcast SQL queries via Maps that
+// contains JSON serialized values.
+func jsonSerializationExample(client *hazelcast.Client) {
+	sqlService := client.SQL()
+	// Operations using JSON
+	createMappingForCountries(sqlService)
+	populateCountriesWithMap(client)
+	selectAllCountries(sqlService)
+	createMappingForCities(sqlService)
+	populateCities(client)
+	selectCitiesByCountry(sqlService, "US")
+	selectCountriesAndCities(sqlService)
+}
+
+func createMappingForCountries(sqlService sql.Service) {
+	//see: https://docs.hazelcast.com/hazelcast/5.1/sql/mapping-to-maps#json-objects
+	fmt.Println("Creating mapping for countries...")
+	mappingQuery := fmt.Sprintf(`
+        CREATE OR REPLACE MAPPING country (
+			__key VARCHAR,
+			isoCode VARCHAR,
+			country VARCHAR
+		)
+        TYPE IMAP 
+        OPTIONS (
+            'keyFormat' = 'varchar',
+            'valueFormat' = 'json-flat'
+        )
+	`)
+	ignored, err := sqlService.Execute(context.Background(), mappingQuery)
+	if err != nil {
+		panic(err)
+	}
+	defer ignored.Close()
+	fmt.Println("Mapping for countries has been created.")
+	fmt.Println("--------------------")
+}
+
+func populateCountriesWithMap(client *hazelcast.Client) {
+	// see: https://docs.hazelcast.com/hazelcast/5.1/data-structures/creating-a-map#writing-json-to-a-map
+	ctx := context.Background()
+	fmt.Println("Populating 'countries' map with JSON values...")
+	countries, err := client.GetMap(ctx, "country")
+	if err != nil {
+		panic(err)
+	}
+	countries.Put(ctx, "AU", model.CountryAsJson("AU", "Australia"))
```

Comment:
Can you add a comment that says you are ignoring the errors?

Reply:
Instead, I handled related error.